openedx/frontend-app-course-authoring
# feat: assign taxonomy to organizations [FC-0036]

## Description
This PR adds the UI to assign organizations to a Taxonomy.

![manage_orgs](https://github.com/openedx/frontend-app-course-authoring/assets/849463/dfe3dd32-7511-41e4-b7af-df8738aa9173)

## Supporting Information
* Closes:
  * https://github.com/openedx/modular-learning/issues/117
* Depends on:
  * https://github.com/openedx/frontend-app-course-authoring/pull/675

## Testing instructions
1. Go to the taxonomy list page (http://localhost:2001/taxonomies)
1. Click on the `Manage Organizations` menu from an existing taxonomy card
1. You will be prompted with a form to assign Organizations to the taxonomy.
1. Assign an org to the taxonomy and click on `Save`
1. Click on the `Manage Organizations` menu from the taxonomy and check if the org was assigned
1. Use the checkbox to `Assign to all organizations`. The other fields must be disabled. Click on `Save`
1. Click on the `Manage Organizations` menu from the taxonomy and check if the Taxonomy was assigned to all orgs
1. Uncheck the `Assign to all organizations`, leave the Taxonomy without assigned orgs and click on `Save`
1. A dialog will ask for confirmation if you leave a Taxonomy without associated orgs
1. Click on the `Manage Organizations` menu from the taxonomy and check if the Taxonomy has no assigned orgs

The feature can also be accessed from the `Actions` menu on the Taxonomy Details page.

![image](https://github.com/openedx/frontend-app-course-authoring/assets/849463/693d1368-92d1-4a8a-94c0-412aa9cfe6e7)

---
Private-ref:
 - [FAL-3543](https://tasks.opencraft.com/browse/FAL-3543)
